fieldpath/path_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldpath
+
+import (
+	"testing"
+
+	"github.com/kubernetes-sigs/structured-merge-diff/value"
+)
+
+func TestPath(t *testing.T) {
+	table := []struct {
+		name   string
+		fp     Path
+		expect string
+	}{
+		{"basic1", MakePathOrDie("foo", 1), ".foo[1]"},
+		{"basic2", MakePathOrDie("foo", "bar", 1, "baz"), ".foo.bar[1].baz"},
+		{"associative-list-ref", MakePathOrDie("foo", KeyByFields(
+			"a", value.StringValue("b"),
```

Comment:
These look like field to me: name of the field and the value. Shouldn't the parameter of `KeyByFields` by a `[]Field`?

Reply:
The point is that this way of specification cuts out a lot of boilerplate.

Reply:
No strong opinion.